content/en/docs/concepts/configuration/manage-resources-containers.md

```diff
@@ -10,23 +10,46 @@ feature:
 
 {{% capture overview %}}
 
-When you specify a [Pod](/docs/concepts/workloads/pods/pod/), you can optionally specify how
-much CPU and memory (RAM) each Container needs. When Containers have resource
-requests specified, the scheduler can make better decisions about which nodes to
-place Pods on. And when Containers have their limits specified, contention for
-resources on a node can be handled in a specified manner. For more details about
-the difference between requests and limits, see
-[Resource QoS](https://git.k8s.io/community/contributors/design-proposals/node/resource-qos.md).
+When you specify a {{< glossary_tooltip term_id="pod" >}}, you can optionally specify how
+much of each resource a {{< glossary_tooltip text="Container" term_id="container" >}} needs.
+The most common resources to specify are CPU and memory (RAM); there are others.
+
+When you specify the resource _request_ for Containers in a Pod, the scheduler uses this
+information to decide which node to place the Pod on. When you specify a resource _limit_
```

Comment:
What kind of guarantee should we document? Is that the kind of guarantee that people can _rely_ on the kubelet fulfilling?

My understanding was that the strength of that guarantee depends on your choice of container runtime.

Reply:
The CRI uses cpu shares for both windows and linux: https://github.com/kubernetes/cri-api/blob/master/pkg/apis/runtime/v1alpha2/api.proto#L689.  Shares should "guarantee" a certain level of access.  Of course, it can't really ever guarantee anything, since kernel mechanisms aren't always perfect... But it is essentially what it is doing.

Reply:
I don't think this document should say guarantee as it might mislead readers into thinking that this mechanism alone is suitable for hard realtime workloads (“Ah, I should set request and limit to 1.0 vCPU and then my workload runs undisturbed”).
Similarly, a request of 512 MiB does not  by itself mean that the memory will absolutely, definitely be there when you call `brk()`.

Maybe there's wording short of guarantee that works better? I'll have a think.

Reply:
I'll leave it up to you to decide wording.  Whatever you prefer works for me.